CMakeLists.txt

```diff
@@ -25,14 +27,22 @@ option(ENABLE_PARAVIEW_CATALYST
 option(ENABLE_TIOGA "Use TIOGA TPL to perform overset connectivity" OFF)
 option(ENABLE_OPENMP "Enable OpenMP flags" OFF)
 option(ENABLE_BOOST  "Enable Boost libraries" OFF)
-option(ENABLE_MATRIXFREE "Enable high-order matrix-free computation" ON)
-option(NALU_WIND_SAVE_GOLDS  "Save gold files to directory when running tests" OFF)
+option(ENABLE_MATRIXFREE "Enable high-order matrix-free computation" OFF)
```

Comment:
I think we should leave this defaulted to on or else it might go stale

Reply:
Yeah, I envisioned it more as a [dependent option](https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html) on by default with tpetra on and always off if tpetra is off.

Reply:
Yes, I can tweak that... the complication is, it can't be on if trilinos-solvers is off, and there's no spack-spec mechanism for flipping it...

Reply:
@rcknaus I can adjust the CMake file to leave matrix-free on by default, and turn it off if trilinos-solvers are off. That's easier than adding the spack mechanism.

Reply:
@alanw0 that works